client/src/tasks_sidebar.ts

```diff
@@ -0,0 +1,395 @@
+    this.contextValue = "noscripts";
+  }
+}
+
+class DenoTaskProvider implements TaskProvider {
+  #extensionContext: DenoExtensionContext;
+
+  constructor(extensionContext: DenoExtensionContext) {
+    this.#extensionContext = extensionContext;
+  }
+
+  async provideTasks(): Promise<Task[]> {
+    const tasks: Task[] = [];
+
+    const process = await getDenoCommandName();
+
+    // we retrieve config tasks from the language server, if the language server
+    // supports the capability
+    const client = this.#extensionContext.client;
+    const supportsConfigTasks = this.#extensionContext.serverCapabilities
+      ?.experimental?.denoConfigTasks;
+    if (client && supportsConfigTasks) {
+      try {
+        const configTasks = await client.sendRequest(taskReq);
+        if (configTasks) {
+          for (const workspaceFolder of workspace.workspaceFolders ?? []) {
+            // Check if config file is .json or .jsonc
+            let fileName = "deno.json";
+            try {
+              const filePath = path.join(workspaceFolder.uri.fsPath, fileName);
+              fs.accessSync(filePath, fs.constants.F_OK);
+            } catch {
+              fileName = "deno.jsonc";
+            }
+            for (const { name, detail: command } of configTasks) {
+              tasks.push(
+                buildDenoConfigTask(
+                  workspaceFolder,
+                  process,
+                  name,
+                  fileName,
+                  command,
+                ),
+              );
+            }
+          }
+        }
+      } catch (err) {
+        window.showErrorMessage("Failed to retrieve config tasks.");
+        this.#extensionContext.outputChannel.appendLine(
+          `Error retrieving config tasks: ${err}`,
+        );
+      }
+    }
+
+    return tasks;
+  }
+
+  // deno-lint-ignore require-await
+  async resolveTask(task: Task): Promise<Task | undefined> {
+    return task;
+  }
+}
+
+type TaskTree = Folder[] | DenoJSON[] | NoScripts[];
+
+export class DenoTasksTreeDataProvider implements TreeDataProvider<TreeItem> {
+  #taskTree: TaskTree | null = null;
+  #extensionContext: DenoExtensionContext;
+  #onDidChangeTreeData = new EventEmitter<TreeItem | null>();
+  readonly onDidChangeTreeData = this.#onDidChangeTreeData.event;
+
+  constructor(
+    context: DenoExtensionContext,
+    public taskProvider: DenoTaskProvider,
+    subscriptions: ExtensionContext["subscriptions"],
+  ) {
+    this.#extensionContext = context;
+    subscriptions.push(
+      commands.registerCommand("deno.task.run", this.#runTask, this),
+    );
+    subscriptions.push(commands.registerCommand(
+      "deno.task.debug",
+      this.#debugTask,
+      this,
+    ));
+    subscriptions.push(commands.registerCommand(
+      "deno.task.definition.open",
+      this.#openTaskDefinition,
+      this,
+    ));
+    subscriptions.push(commands.registerCommand(
+      "deno.tasks.refresh",
+      this.refresh.bind(this),
+    ));
```

Comment:
I think we should avoid granular namespaces for command names, can we rename these to the following?
- `deno.client.runTask`
- `deno.client.debugTask`
- `deno.client.openTaskDefinition`
- `deno.client.refreshTasks`

Reply:
Ok, I've made these changes.